addon/utils/_private/rdfa-utils.ts

```diff
@@ -179,3 +173,47 @@ export function getResources(node: PNode): Set<string> {
   });
   return result;
 }
+
+export function findRdfaIdsInSelection(selection: Selection) {
+  const result = new Set<string>();
+  const range = selection.$from.blockRange(selection.$to);
+  if (!range) return result;
+  const commonParent = range.parent;
+
+  commonParent.nodesBetween(
+    range.$from.pos - range.start,
+    range.$to.pos - range.start,
+    (child) => {
+      const id = getRdfaId(child);
+      if (id) {
+        result.add(id);
+      }
+      return true;
+    },
+  );
+  return result;
+}
+
+/**
+ * Get all of the literal nodes which are children of the given node
+ */
+export function getChildLiterals(node: PNode) {
```

Comment:
Seems to me this generates external properties for all descendant rdfa-aware nodes (not just literals, also resource nodes)
If intentional, the function could perhaps use a better name and docstring

Reply:
Hmm, I was thinking that we wouldn't want to automatically link a wrapping resource to child resources, just literals, but I'm not sure what makes most sense now. Maybe this is one to decide after actually using the editor to make a big template. I'll at least ake the name and doc consistent for now.